soccer/joystick/GamepadController.cpp

```diff
@@ -0,0 +1,178 @@
+#include "GamepadController.hpp"
+
+using namespace std;
+
+namespace {
+constexpr RJ::Time Dribble_Step_Time = 125 * 1000;
+constexpr RJ::Time Kicker_Step_Time = 125 * 1000;
+const float AXIS_MAX = 32768.0f;
+}
+
+GamepadController::GamepadController()
+    : _controller(nullptr), _lastDribblerTime(0), _lastKickerTime(0) {
+
+    // initialize using the SDL joystick
+    if (SDL_Init(SDL_INIT_GAMECONTROLLER) != 0) {
+        cerr << "ERROR: SDL could not initialize game controller system! SDL Error: " << SDL_GetError()
+             << endl;
+        return;
+    }
+
+    if (SDL_NumJoysticks()) {
+        // Open the first available controller
+        for (size_t i = 0; i < SDL_NumJoysticks(); ++i) {
+            // // open joystick, print out info about it, then close it back
+            // SDL_Joystick *joy;
+            // joy = SDL_JoystickOpen(i);
+            // if (joy) {
+            //     cerr << "Joystick " << i << ":" << endl;
+            //     cerr << "  Name:\t\t\t" << SDL_JoystickNameForIndex(i) << endl;
+            //     cerr << "  Number of Axes:\t" << SDL_JoystickNumAxes(joy) << endl;
+            //     cerr << "  Number of Btns:\t" << SDL_JoystickNumButtons(joy) << endl;
+            //     cerr << "  Number of Balls:\t" << SDL_JoystickNumBalls(joy) << endl;
+            //     cerr << "  Game Controller:\t" << static_cast<bool>(SDL_IsGameController(i)) << endl;
+            // } else {
+            //     cerr << "Unable to open joystick number " << i << endl;
+            // }
+
+            // // close the joystick back
+            // if (SDL_JoystickGetAttached(joy)) {
+            //     SDL_JoystickClose(joy);
+            // }
```

Comment:
I think this was code relevant to the old controller, but not the new one? Since we have the old controller in use again we should be able to delete this (I think).